runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp

```diff
@@ -1,404 +1,318 @@
 
-    if (1 != network_groups->size()) {
-        std::cerr << "Invalid amount of network groups" << std::endl;
-        return make_unexpected(HAILO_INTERNAL_FAILURE);
-    }
-
-    return std::move(network_groups->at(0));
+    std::cout << "\n-I- Inference finished successfully\n" << std::endl;
+    return HAILO_SUCCESS;
 }
 
-std::string getCmdOption(int argc, char *argv[], const std::string &option)
-{
-    std::string cmd;
-    for (int i = 1; i < argc; ++i)
-    {
-        std::string arg = argv[i];
-        if (0 == arg.find(option, 0))
-        {
-            std::size_t found = arg.find("=", 0) + 1;
-            cmd = arg.substr(found);
-            return cmd;
-        }
-    }
-    return cmd;
-}
 
 int main(int argc, char** argv) {
-
-    hailo_status status = HAILO_UNINITIALIZED;
-
-    std::chrono::duration<double> total_time;
-    std::chrono::time_point<std::chrono::system_clock> t_start = std::chrono::high_resolution_clock::now();
-
-    std::string yolo_hef       = getCmdOption(argc, argv, "-hef=");
-    std::string video_path      = getCmdOption(argc, argv, "-input=");
-
-    std::chrono::time_point<std::chrono::system_clock> write_time_vec;
-    std::chrono::duration<double> inference_time;
-    std::chrono::duration<double> postprocess_time;
-
-    auto vdevice_exp = VDevice::create();
-    if (!vdevice_exp) {
-        std::cerr << "Failed create vdevice, status = " << vdevice_exp.status() << std::endl;
-        return vdevice_exp.status();
+    std::string hef_file   = getCmdOption(argc, argv, "-hef=");
+    std::string video_path = getCmdOption(argc, argv, "-path=");
+    auto all_devices       = Device::scan_pcie();
+    std::cout << "-I- video path: " << video_path << std::endl;
+    std::cout << "-I- hef: " << hef_file << "\n" << std::endl;
+
+    auto device = Device::create_pcie(all_devices.value()[0]);
+    if (!device) {
+        std::cerr << "-E- Failed create_pcie " << device.status() << std::endl;
+        return device.status();
     }
-    auto vdevice = vdevice_exp.release();
 
-    auto network_group = configure_network_group(*vdevice, yolo_hef);
+    auto network_group = configure_network_group(*device.value(), hef_file);
     if (!network_group) {
-        std::cerr << "Failed to configure network group " << yolo_hef << std::endl;
+        std::cerr << "-E- Failed to configure network group " << hef_file << std::endl;
         return network_group.status();
     }
 
-    auto input_vstreams_params = network_group.value()->make_input_vstream_params(QUANTIZED, FORMAT_TYPE_INPUT, HAILO_DEFAULT_VSTREAM_TIMEOUT_MS, HAILO_DEFAULT_VSTREAM_QUEUE_SIZE);
-    if (!input_vstreams_params) {
-        std::cerr << "Failed creating input vstreams " << input_vstreams_params.status() << std::endl;
-        return input_vstreams_params.status();
+    auto input_vstream_params = network_group.value()->make_input_vstream_params(true, HAILO_FORMAT_TYPE_UINT8, HAILO_DEFAULT_VSTREAM_TIMEOUT_MS, HAILO_DEFAULT_VSTREAM_QUEUE_SIZE);
```

Comment:
I think that you can use HAILO_FORMAT_TYPE_AUTO for both input and output and then you don't have to deal with HEFs that are with different data format (which shouldn't happen, but who knows what would be in the future).